src/screens/main.screen.tsx

```diff
@@ -45,8 +46,13 @@ const MainScreen = ({ navigation }: ScreenProps) => {
   }, [location]);
 
   useEffect(() => {
-    if (locationError) {
-      navigation.navigate(SCREENS.NoLocation);
+    if (locationError && !navigation.canGoBack()) {
```

Comment:
Seems to work perfectly, but what is the idea with `!navigation.canGoBack()` here?

Reply:
In the case that the no location screen loads the first time and you click one of the districts, you'd go to the main page but the main page wouldd trigger another location error and take you back to the no location screen. Like a loop. The additional condition checks that there's no history and stops that behaviour. So it errors and takes you to the no location screen the first time but does not subsequently.